ir/attrs.h

```diff
@@ -69,4 +73,12 @@ class FnAttrs final {
   friend std::ostream& operator<<(std::ostream &os, const FnAttrs &attr);
 };
 
+
+// Encode attributes for a non-aggregate value using UB and poison.
+// Attributes requiring more than UB or poison are not encoded.
+void encodeParamAttrs(const ParamAttrs &attrs, State &s, StateValue &value,
```

Comment:
can we move these to the respective classes?

TBH, I don't really like these functions changing StateValue directly. It's very non-obvious. Can we return a pair with the UB (AndExpr) and poison (maybe expr is enough)?